File: 6KYU/Backspaces in string/solution.cs
public class Kata
{
  public static string CleanString(string s)
  {
    var a = s.ToCharArray();
    var result = string.Empty;
    var charsToSkip = 0;  
    for(int i = a.Length-1;  i >= 0; i--)
    {
      if(a[i] == '#')
        charsToSkip ++;
      else
      {
        if(charsToSkip > 0)
          charsToSkip --;
        else
          result = a[i].ToString() + result;
      }    
    }
    return result;
  }
}

File: 6KYU/Backspaces in string/solution.go
package kata

func CleanString(s string) string {
	res := ""
	length := len(s)
	charsToSkip := 0
	for i := length - 1; i >= 0; i-- {
		if s[i] == '#' {
			charsToSkip = charsToSkip + 1
		} else {
			if charsToSkip > 0 {
				charsToSkip = charsToSkip - 1
			} else {
				res = string(s[i]) + res
			}

		}
	}
	return res
}
